src/pages/Code.tsx

```diff
@@ -1,15 +1,107 @@
-import React, { useState } from 'react'
+import React, { useEffect, useLayoutEffect, useState } from "react";
+import BoxContainer from "../components/BoxContainer";
+import AppLayout from "../layouts/AppLayout";
+import { Text, Box, Center, Flex, Image } from "@chakra-ui/react";
 
 type code = {
-    code: string ,
-    end: Date
-}
+  code: string;
+  end: Date;
+};
 
 const Code = () => {
-    const [code] = useState<code>({code:"X23AD",end: new Date(new Date().getTime() + 1*60000)})
+  const [code, setCode] = useState<code>({
+    code: "X23AD",
+    end: new Date(new Date().getTime() + 1 * 60000),
+  });
+
+  const countdownDate = code.end.getTime();
+  const countdown = () => {
+    const now = new Date().getTime();
+    const timeLeft = countdownDate - now;
+    let seconds = Math.floor((timeLeft % (1000 * 60)) / 1000);
+    return seconds;
+  };
+  const [time, setTime] = useState(countdown());
+
+  useLayoutEffect(() => {
+    setTime(countdown());
+  }, []);
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      if (countdown() <= 0) {
```

Comment:
if implemented with dayjs, you can just use `.isBefore()` to check instead of `countdown()`